rbc/heavydb/row_function_manager.py

```diff
@@ -68,7 +72,7 @@ def heavydb_udf_manager_get_dict_id(mgr, func_name, arg_idx):
     if target_info.software[1][:3] < (6, 2, 0):
         raise UnsupportedError(error_msg % (".".join(map(str, target_info.software[1]))))
 
-    defn = 'int32 RowFunctionManager_getDictId(int8*, int8*, int32)'
+    defn = 'int32 RowFunctionManager_getDictId(int8*, int8*, int32)|CPU'
```

Comment:
Could we move the device specification to a keyword argument? Similar to what we do with the `@heavydb` decorator? i.e.:

```python
defn = 'int32 RowFunctionManager_getDictId(int8*, int8*, int32)'
get_dict_id = external(defn, devices=['cpu'])
```

Reply:
I think this makes sense. We should also get rid of the `| CPU` annotation as it duplicates the `devices` kw argument. What do you think?

Reply:
Agreed. It can be done in a follow-up PR.

Reply:
Done: introduced `devices` kw argument to `external` and deprecated CPU/GPU annotations.